app/controllers/app/role/page.js

```diff
@@ -38,4 +48,68 @@ export default class AppRolePageController extends AppRoleController {
             });
         }
     }
+
+    /**
+     * This method returns the role permissions by creating the associative array containing the key
+     * as the module name and there permissions.
+     * 
+     * @property permissions
+     * @protected
+     */
+    @computed('model.permissions')
+    get permissions() {
+        let rolesPermissions = {};
+        let permissions = this.model.permissions;
+        let modelsList = permissions.reduce((models, permission) => {
+            if (!permission.resourceName.includes('.')) {
+                models.push(permission.resourceName);
+            }
+            return models;
+        }, []);
+
+
+        //add childs (fields, relationships) inside the parent.
+        modelsList.forEach((modelName) => {
+            let modelPermissions = permissions.filter((permission) => {
+                let [model, fieldName] = permission.resourceName.split('.');
+                if (modelName === model) {
+                    permission.resourceName = fieldName ?? model;
+                    return permission;
+                }
+            });
+
+            rolesPermissions[modelName] = modelPermissions;
+        });
+        return rolesPermissions;
+    }
+
+    /**
+     * This returns the all of the permission flags.
+     * 
+     * @property permissionFlags
+     * @protected
+     */
+    get permissionFlags() {
```

Comment:
I think this information should be returned by the API, we already have this info in the meta.